layouts/partials/sidebar.html

```diff
@@ -5,9 +5,16 @@
   <ul class="nav flex-column">
     {{ range $menu }}
       <li class="nav-item">
-        <a class="nav-link {{ if eq .Page.RelPermalink $current }}active{{ end }}" href="{{ .Page.RelPermalink }}">
+        {{ $isCurrent := and .Page (eq .Page.RelPermalink $.RelPermalink) }}
+        <a class="nav-link {{ if $isCurrent }}active{{ end }}" href="{{ if .Page }}{{ .Page.RelPermalink }}{{ else }}{{ .URL }}{{ end }}">
           {{ .Name }}
         </a>
+
+        {{ if and $isCurrent .Page.Params.show_section_nav }}
+          <div class="sidebar-toc nav flex-column ms-3 fs-6">
```

Comment:
Would be better to use `ms-4`, so the sub sections will be indented slightly inwards of the main menu.

Also I think `$current`  in line 2 is not relevant anymore and maybe we can we remove it.